### PR TITLE
Flesh out API structure. 

### PR DESCRIPTION
**Pull request by zelnaut**
Setup basic API structure:
- Add logging library (zap)
- Switch to Viper for config management
- Use Swag for OpenAPI docs
- Add Prometheus for metrics
- Use Makefile to build Go project

Might want to add Prometheus UI or Grafana to our docker setup down the line.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,11 +1,22 @@
 {
   "editor.formatOnSave": true,
+  "files.eol": "\n",
```
Comment (zelnaut) [inline .vscode/settings.json:3]: This should prevent that Prettier error from happening again

File: api/main.go [hunk 2]
```diff
@@ -1,115 +1,44 @@
...
-
-	userInfo := &User{
-		Name:     user.Name,
-		Password: user.Password,
-	}
-
-	res, err := db.NewInsert().Model(userInfo).Exec(context.Background())
-	if err != nil {
-		fmt.Printf("Insert error: %v", err)
+// NOTE: We might be better storing "globals" in the Gin context instead.
```
Comment (zelnaut) [inline api/main.go:15]: Need to look into this a little more, but using globals might make it a little hard to test this.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -34,18 +34,16 @@ jobs:
...
-          git diff --exit-code go.mod
+        run: make lint-ci
 
       - name: Build API
-        run: go build main.go
+        run: make build
 
   web:
-    name: Web CI
+    name: Web
```
Comment (QuickFinesse) [inline .github/workflows/ci.yml:46]: 👍

File: .vscode/settings.json (hunk 1 above)
Comment (QuickFinesse) [inline .vscode/settings.json:3]: 👍 good add, thanks for looking into this

File: scripts/up.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+docker compose up --build --detach
```
Comment (QuickFinesse) [inline scripts/up.sh:3]: I love these script files, I'm writing documentation now and it's a lot easier to tell people to just run a simple script lolol, thanks for this 🙌

**Review by QuickFinesse — APPROVED** [review]
Works perfectly, one question, if I direct people wanting to help on the project to run the bash script you created, how are these supposed to see the api outputs/ errors? I'm probably overlooking something but just wondering how they'd see that

Maybe by default we include the logs? Like the script you wrote earlier -->  `docker compose down --volumes --remove-orphans && docker compose up --build --detach && docker compose logs --follow` ? Dunno 🤷

File: api/main.go (hunk 2 above)
Comment (QuickFinesse) [reply api/main.go:15]: Let's talk about this some more, don't really understand what you mean with regards to "globals"

**Comment by zelnaut** [comment]
> how are these supposed to see the api outputs/ errors
Might be easier to see the logs through docker desktop than the terminal.

File: api/main.go (hunk 2 above)
Comment (zelnaut) [reply api/main.go:15]: Globals being variables that exist in the global scope, so in this PR I added the DB connection as a global.

```
var Database *bun.DB
```

Globals usually aren't recommended, but they seem fairly popular in Go. Not sure what the correct way to manage things like the DB if we're not using globals. Usually the framework you're using will provide a way to manage them, through a form of dependency injection.

Comment (QuickFinesse) [reply api/main.go:15]: Ahhh I see, yeah I used them before but good question. I think as we progress/ both get better with Go we can refactor and make this layout make more sense. Globals seem weird but if Go uses em 🤷

**Pull request merged**